FinnUI/Sources/Fullscreen/Settings/SettingsView.swift

```diff
@@ -0,0 +1,181 @@
+public struct SettingsView: View {
+    private let sections: [SettingsSection]
+
+    public init(sections: [SettingsSection]) {
+        self.sections = sections
+    }
+
+    public var body: some View {
+        List {
+            ForEach(0..<sections.count) { section in
+                if self.sections[section].title != nil {
+                    Header(text: self.sections[section].title!)
+                }
+
+                ForEach(0..<self.sections[section].items.count) { row in
+                    self.cell(at: row, in: section)
+                        .bottomDivider(row != self.sections[section].items.count - 1)
+                }
+
+                if self.sections[section].footerTitle != nil {
+                    Footer(text: self.sections[section].footerTitle!)
+                }
+            }
+        }
+        .listSeparatorStyleNone()
+        .edgesIgnoringSafeArea(.all)
+    }
+
+    private func cell(at row: Int, in section: Int) -> AnyView {
+        let model = sections[section].items[row]
+
+        switch model {
+        case let model as SettingsViewToggleCellModel:
+            return AnyView(ToggleCell(model: model))
+        case let model as SettingsViewConsentCellModel:
+            return AnyView(ConsentCell(model: model))
+        default:
+            return AnyView(BasicListCell(model: model))
+        }
+    }
+}
+
+// MARK: - Cells
+
+@available(iOS 13.0.0, *)
+private struct Header: View {
+    let text: String
+
+    var body: some View {
+        VStack {
+            Spacer()
+            HStack {
+                Text(text.uppercased())
+                    .font(Font(UIFont.detailStrong))
+                    .foregroundColor(.textSecondary)
+                    .padding(.horizontal, .spacingM)
+                    .padding(.bottom, .spacingS)
+                Spacer()
+            }
+        }
+        .listRowInsets(EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0))
+        .background(Color.bgTertiary)
+        .frame(height: 48)
+    }
+}
+
+@available(iOS 13.0.0, *)
+private struct Footer: View {
+    let text: String
+
+    var body: some View {
+        HStack {
+            Text(text)
+                .font(Font(UIFont.caption))
+                .foregroundColor(.textSecondary)
+                .padding(.horizontal, .spacingM)
+                .padding(.vertical, .spacingS)
+            Spacer()
+        }
+        .background(Color.bgTertiary)
+        .listRowInsets(EdgeInsets())
+    }
+}
+
+@available(iOS 13.0.0, *)
+private struct ConsentCell: View {
+    let model: SettingsViewConsentCellModel
+
+    var body: some View {
+        BasicListCell(model: model, detailText: { _ in
+            Text(model.status)
+                .font(Font(UIFont.body))
+                .foregroundColor(.textSecondary)
+        })
+    }
+}
+
+@available(iOS 13.0.0, *)
+private struct ToggleCell: View {
+    let model: SettingsViewToggleCellModel
+    @SwiftUI.State private var isOn = true
```

Comment:
Why the SwiftUI prefix? is there a type that crashes?

Reply:
I don't know. For some reason I'm getting "Unknown attribute 'State'" compiler error if `SwiftUI` module name is not specified. Can't find any conflicting types in Finnivers though.